lib/topolys/model.rb

```diff
@@ -948,50 +948,68 @@ def recalculate
           @normal = temp
         end
       end
-      
+
       raise "Cannot compute normal" if @normal.nil?
       raise "Normal has 0 length" if largest == 0
-      
+
       @normal.normalize!
-      
+
       @plane = Topolys::Plane3D.new(@directed_edges[0].v0.point, @normal)
-    
+
+      # Recourse to Float::EPSILON occurs in 4x places in model.rb. Direct
+      # comparison between Float::EPSILON (i.e. 2.2204460492503131e-16)
+      # and something like a distance (in millimeters) is problematic. If
+      # epsilon were instead a relative tolerance (one could be more stringent
+      # when assessing a 8cm x 8cm surface instead of a 8m x 8m surface), then
+      # that could work. But for now, I think we're OK with the absolute 'tol'
+      # of 10mm (some days I think this could be stretched to 25mm) for
+      # vertex equivalence, is a vertex part of a line or a plane, etc.
+
+      # If I replace all occurrences of Float::EPSILON with 0.01, all
+      # raises/errors go away. I attempted to use @tol, but this raised
+      # a 'comparison of Float with nil failed', which I think is related to
+      # the Model initialize method. To re-assess?
+
       @directed_edges.each do |de|
-        raise "Point not on plane" if (de.v0.point - @plane.project(de.v0.point)).magnitude > Float::EPSILON
-        raise "Point not on plane" if (de.v1.point - @plane.project(de.v1.point)).magnitude > Float::EPSILON
+        raise "Point not on plane - 2" if (de.v0.point - @plane.project(de.v0.point)).magnitude > 0.01 #Float::EPSILON
```

Comment:
@rd2 this does seem like a better tolerance, we should make it into a configuration somewhere